android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/LoginFragment.kt

```diff
@@ -0,0 +1,130 @@
+                    Log.w(TAG, "Google sign in failed", e)
+                }
+            } else {
+                Log.w(TAG, "Google sign in unsuccessful")
+            }
+
+        }
+        // No other requestCode, ignore it.
+    }
+
+
+    private fun firebaseAuthWithGoogle(idToken: String) {
+        Log.d(TAG, "firebaseAuthWithGoogle start")
+        val credential = GoogleAuthProvider.getCredential(idToken, null)
+        Log.d(TAG, "firebaseAuthWithGoogle credential:$credential")
+        signInVM.getFirebaseAuth().signInWithCredential(credential)
+            .addOnCompleteListener { task ->
+                Log.d(TAG, "firebase firebaseAuthWithGoogle:task check")
+                if (task.isSuccessful) {
+                    // Firebase Sign in success, update UI with the signed-in user's information
+                    Log.d(TAG, "firebase signInWithCredential:success")
+                    val user = signInVM.getFirebaseAuth().currentUser
+                    Log.d(TAG, "firebase signed-in user's Email:" + user!!.email)
+                } else {
+                    // If sign in fails, log a message to the user.
+                    Log.w(TAG, "signInWithCredential:failure", task.exception)
+                }
+            }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            param1 = it.getString(ARG_PARAM1)
+            param2 = it.getString(ARG_PARAM2)
+        }
+
+        signInVM = activity?.run {
+            ViewModelProviders.of(this)[SignInViewModel::class.java]
+        } ?: throw Exception("Null Activity")
+
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_login, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        view.findViewById<SignInButton>(R.id.sign_in_button).setOnClickListener {
+            signIn()
+        }
+    }
+
+    companion object {
+        /**
+         * Use this factory method to create a new instance of
+         * this fragment using the provided parameters.
+         *
+         * @param param1 Parameter 1.
+         * @param param2 Parameter 2.
+         * @return A new instance of fragment LoginFragment.
+         */
+        const val RC_SIGN_IN = 0
+        const val TAG = "Login fragment"
+        // TODO: Rename and change types and number of parameters
+        @JvmStatic fun newInstance(param1: String, param2: String) =
+                LoginFragment().apply {
+                    arguments = Bundle().apply {
+                        putString(ARG_PARAM1, param1)
+                        putString(ARG_PARAM2, param2)
+                    }
```

Comment:
Skip this and just return the new fragment instance if you don't need the arguments.